cpp/src/fil/infer.cu

```diff
@@ -424,7 +423,8 @@ __global__ void infer_k(storage_type forest, predict_params params) {
       }
       if (leaf_algo == GROVE_PER_CLASS_MANY_CLASSES) __syncthreads();
     }
-    acc.finalize(params.preds, params.num_rows, params.num_outputs);
+    acc.finalize(params.preds + params.num_outputs * grid_row0, params.num_rows,
```

Comment:
no, because in the unlimited-blocks mode, the `finalize()` code computes the offset into params.preds with `params.num_outputs * blockIdx.x * NITEMS`, since `blockIdx.x * NITEMS` is the row number. Hence, I need to pass something that does not already contain `blockIdx.x * NITEMS` as a summand.